src/main/java/com/minecolonies/entity/EntityFishHook.java

```diff
@@ -748,21 +712,40 @@ private ItemStack getFishingLoot(@NotNull final EntityCitizen citizen)
 
         if (random < speedBonus || buildingLevel == 1)
         {
-            return ((WeightedRandomFishable) WeightedRandom.getRandomItem(this.rand, fishDrops)).getItemStack(this.rand);
+            LootContext.Builder lootcontext$builder = new LootContext.Builder((WorldServer) this.worldObj);
+            for (ItemStack itemstack : this.worldObj.getLootTableManager()
+                                         .getLootTableFromLocation(LootTableList.GAMEPLAY_FISHING)
+                                         .generateLootForPools(this.rand, lootcontext$builder.build()))
+            {
+                return itemstack;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Avoid using a branching statement as the last in a loop. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AAvoidBranchingStatementAsLastInLoop)